src/app/hooks/useScrolling.ts

```diff
@@ -0,0 +1,28 @@
+import { useLocation } from 'react-router-dom'
+import { useEffect, useRef } from 'react'
+
+/**
+ * React hook that take care of (fixes) automatic scrolling.
+ *
+ * Normally, if an HTML element has an anchor, then it can be added to the URL, and we can scroll to this div.
+ *
+ * However, when loading the page initially, sometimes the HTML element for the card is not yet available in the DOM
+ * when the scrolling is supposed to happen, so it fails. This hook can fix that.
+ *
+ * Use this hook to implement scrolling. See the ScrollingCard component for an example on how to use.
+ */
+export function useScrolling<ElementType extends Element = Element>(targetHash: string | undefined) {
```

Comment:
Should these docs be changed too?

Without looking at ScrollingCard, I would not know this is intended to be passed into `ref=`, but I like the abstraction